README
You can install this using in the usual Perl fashion

	perl Makefile.PL
	make
	make test
	make install
	
The documentation is in the module file.  Once you install
the file, you can read it with perldoc.

	perldoc HTTP::SimpleLinkChecker
	
If you want to read it before you install it, you can use
perldoc directly on the module file.

	perldoc lib/SimpleLinkChecker.pm
	
This module is also in Github:

	http://github.com/briandfoy/http-simplelinkchecker


Enjoy, 

brian d foy, bdfoy@cpan.org


Changes
Revision history for Perl extension HTTP::SimpleLinkChecker.

1.16 - Sun May 23 13:21:12 2010
	* Fix 403 status test (Google now returns 502)
	* Moved everything to Github

1.15 - Sat Dec  8 20:26:44 2007
	* add the http_status command line program to check a URL (and the 
	links it contains)
	* require 5.006 now
	* otherwise, things are the same as before :)

1.14 - Sat Nov 24 05:08:50 2007
	* Cleanups for the move from CVS to SVN
	* added teh httpstatus command line program to make it even
	easier

1.13 - Tue Jan  9 22:23:35 2007
	* updated copyright and license info
	* no code changes, so no need to upgrade

1.11 - Mon Dec 25 18:20:44 2006
	* Somehow I forgot to include Exporter, so the exports weren't working.
	Now they are.

1.10 - Thu Jun  8 07:54:27 2006
	* Distro cleanup; no need up to upgrade

1.09 - Thu Jul  7 15:32:25 2005
	* Updated docs to note the things that can go wrong when checking links.
	There is no need to upgrade

1.08 - Fri Mar 11 23:24:44 2005
	* Added POD coverage tests, fixed issues
	* no code changes: no need to upgrade

1.07 - Wed Jan 26 09:14:43 2005
	* I now detect proxy settings with LWP::UserAgent's env_proxy().  This
	fixes CPAN RT #11069.

1.06 - Tue Sep  7 21:25:39 2004
	* cleaned up distro
	* added tests and now use Test::More
	* no need to upgrade if you already have this module

1.5 Sun Jun 23 11:27:43 CDT 2002
	* the user-agent variable is now private, although you
	can get at it with a method

0.51 Wed Jan 24 14:36:53 EST 2001
	* minor doc fixes

0.01  Sat Oct  7 23:13:13 2000
	* original version; created by h2xs 1.18



lib/SimpleLinkChecker.pm
package HTTP::SimpleLinkChecker;
use strict;

use warnings;
no warnings;

use base qw(Exporter);
use vars qw($ERROR $VERSION @EXPORT_OK);

use HTTP::Request;
use LWP::UserAgent;

@EXPORT_OK = qw(check_link);

my $UA = LWP::UserAgent->new();
$UA->env_proxy;

$VERSION = '1.16';

sub check_link
	{
	my $link = shift;
	unless( defined $link )
		{
		$ERROR = 'Received no argument';
		return;
		}

	my $request = HTTP::Request->new('HEAD', $link);
	unless( ref $request )
		{
		$ERROR = 'Could not create HEAD request';
		return;
		}

	my $response = $UA->request($request);

	if( ref $response and $response->code >= 400 )
		{
		$request = HTTP::Request->new('GET', $link);
		unless( ref $request )
			{
			$ERROR = 'Could not create GET request';
			return;
			}
		$response = $UA->request($request);
		}

	unless( ref $response )
		{
		$ERROR = 'Could not get response';
		return;
		}

	return $response->code;
	}

sub user_agent
	{
	return $UA;
	}

1;

=head1 NAME

HTTP::SimpleLinkChecker - Check the HTTP response code for a link

=head1 SYNOPSIS

	use HTTP::SimpleLinkChecker;

	my $code = HTTP::SimpleLinkChecker::check_link($url);

	unless( defined $code )
		{
		print "Error: $HTTP::SimpleLinkChecker::ERROR\n";
		}

=head1 DESCRIPTION

You don't have to know anything about objected-oriented Perl, LWP, or
the HTTP module to be able to check your links. This module is
designed for the casual user. It has one function, C<check_link>, that
returns the HTTP response code that it receives when it tries to fetch
the web address passed to it. The undef value is returned for any
non-HTTP failure and the C<$HTTP::SimpleLinkChecker::ERROR> variable
is set.

The HEAD method is tried first, although if anything other than a good
status code (those less than 400) is received, another request is made
with the GET method. Note, however, that even with the best code, no
module can control how servers decide to respond to a check, or
control any of the myriad things that can go wrong with the network
between you and the remote server. Some may filter requests based on
origin IP address, user-agent type, or any other arbitrary factor.
Some servers may not respond correctly at all. Furthermore, some
servers might be temporarily down or overloaded. I recommend that you
recheck "broken" links a couple times over a long period (like a day
or two) before you decide they are really broken.

If you are behind a firewall or proxy, this module picks up those
settings through LWP::UserAgent's env_proxy() method.  See
L<LWP::UserAgent> for more details.

=head2 Functions

=over 4

=item check_link( URL )

Returns the HTTP response code for URL.

=item user_agent

Returns a reference to the LWP::UserAgent object.  You
can affect it directly.  See L<LWP::UserAgent>.

	my $ua = HTTP::SimpleLinkChecker::user_agent();
	$ua->from( 'joe@example.com' );
	$ua->agent( 'Mozilla 19.2' );

=back

=head1 SOURCE AVAILABILITY

This source is in Github:

	http://github.com/briandfoy/http-simplelinkchecker

If, for some reason, I disappear from the world, one of the other
members of the project can shepherd this module appropriately.

=head1 AUTHOR

brian d foy, C<< <bdfoy@cpan.org> >>

=head1 COPYRIGHT AND LICENSE

Copyright (c) 2004-2010 brian d foy. All rights reserved.

This program is free software; you can redistribute it and/or modify
it under the same terms as Perl itself.

=cut


Makefile.PL
use ExtUtils::MakeMaker;
# See lib/ExtUtils/MakeMaker.pm for details of how to influence
# the contents of the Makefile that is written.

require 5.006;

eval "use Test::Manifest 1.21";

WriteMakefile(
	'NAME'         => 'HTTP::SimpleLinkChecker',
	'ABSTRACT'     => 'Check the HTTP response code for a link',
	'VERSION_FROM' => 'lib/SimpleLinkChecker.pm',
	'LICENSE'      => 'perl',
	'AUTHOR'       => 'brian d foy <bdfoy@cpan.org>',

	'PM'           => {
		'lib/SimpleLinkChecker.pm' => '$(INST_LIBDIR)/SimpleLinkChecker.pm',
		},

	'EXE_FILES'    => [ qw(script/http_status) ],
	
	'PREREQ_PM'    => { 
		'HTTP::Request'  => '1.28',
		'HTTP::Status'   => '0',
		'LWP::UserAgent' => '1.09',
		},
		
	'MAN3PODS'     => {
		'lib/SimpleLinkChecker.pm' => '$(INST_MAN3DIR)/HTTP::SimpleLinkChecker.$(MAN3EXT)',
		},

    clean  => { FILES    => q|HTTP-SimpleLinkChecker-*| },

    (
    ( $ExtUtils::MakeMaker::VERSION ge '6.46' )
        ? (META_MERGE => {
            resources => {
                homepage    => 'http://github.com/briandfoy/http-simplelinkchecker',
                repository  => 'git://github.com/briandfoy/http-simplelinkchecker.git',
            }
        })
        : ()
    )

);
